docs/api.rst

```diff
@@ -4427,6 +4448,25 @@ Slash Options
       - Attachment
       -
 
+.. list-table:: Advanced Annotations
+    :widths: 25 75
+    :header-rows: 1
+
+    * - Typing Annotation
+      - Notes
+    * - :class:`Optional` [<type>]
+      - Makes the slash option not required with the type of the given <type>.
+    * - :class:`Literal` [<choice1>, <choice2>, ...]
+      - Makes the slash option into a choice of the given type. All choices have to be the same data type.
+    * - :class:`Union` [<type1>, <type2>, ...]
+      - Allows compatible annotations to be typehinted together. Channel types in a union will allow the slash option
+        to be any of them.
+    * - :class:`Annotated` [<type1>, <type2>, ..., <typeN>]
+      - Makes your IDE see <type1> as the expected type, but Nextcord will attempt to use <typeN> as the option type.
+        If <typeN> isn't a valid option type, Nextcord will work backwards towards <type2> until it finds a type that
+        can be
+        used as a valid option type.
```

Comment:
```suggestion
        can be used as a valid option type.
```

Reply:
Added commit via web, I/someone will need to build docs and see if something goes wrong.